src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,78 @@
-        return 0;
+        return numbers.stream()
+            .flatMap(stringArray -> Arrays.stream(stringArray.split(",")))
+            .mapToInt(Integer::parseInt)
+            .filter(integer -> integer % 2 == 0)
+            .min()
+            .orElseThrow(() -> new RuntimeException(
+                String.format("Can't get min value from list: < %s >", String.join(" ", numbers))));
     }
 
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+            .map(index -> index % 2 != 0 ? numbers.get(index) - 1 : numbers.get(index))
+            .filter(integer -> integer % 2 != 0)
+            .average()
+            .orElseThrow(NoSuchElementException::new);
     }
 
-    /**
-     * Given a List of `Person` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new Person(«Victor», 16, Sex.MAN),
-     * new Person(«Helen», 42, Sex.WOMAN))`,
-     * select from the List only men whose age is from `fromAge` to `toAge` inclusively.
-     * <p>
-     * Example: select men who can be recruited to army (from 18 to 27 years old inclusively).
-     */
     public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        Predicate<Person> personPredicate = person -> person.getAge() >= fromAge
+                && person.getAge() <= toAge
+                && person.getSex().equals(Sex.MAN);
+        return peopleList.stream()
+            .filter(personPredicate)
+            .collect(Collectors.toList());
     }
 
-    /**
-     * Given a List of `Person` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new Person(«Victor», 16, Sex.MAN),
-     * new Person(«Helen», 42, Sex.WOMAN))`,
-     * select from the List only people whose age is from `fromAge` and to `maleToAge` (for men)
-     * or to `femaleToAge` (for women) inclusively.
-     * <p>
-     * Example: select people of working age
-     * (from 18 y.o. and to 60 y.o. for men and to 55 y.o. for women inclusively).
-     */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
-                                          int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+            int maleToAge, List<Person> peopleList) {
+        Predicate<Person> personPredicate = person -> {
+            if (person.getAge() >= fromAge) {
+                if (person.getSex().equals(Sex.MAN) && person.getAge() <= maleToAge) {
+                    return true;
+                }
+                return person.getAge() <= femaleToAge;
+            }
+            return false;
+        };
```

Comment:
or we can remove if-else constructions
```suggestion
        Predicate<Person> personPredicate = person -> 
            person.getAge() >= fromAge && (person.getSex().equals(Sex.MAN) 
                ? person.getAge() <= maleToAge 
                : person.getAge() <= femaleToAge);
```